src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -1943,15 +1943,17 @@ public static boolean zeroArgs(Object[] args) {
     @CoreMethod(names = "succ!", raiseIfFrozenSelf = true)
     public abstract static class SuccBangNode extends CoreMethodArrayArgumentsNode {
 
-        @TruffleBoundary
+        @Child private RopeNodes.MakeLeafRopeNode makeLeafRopeNode = RopeNodes.MakeLeafRopeNode.create();
+
         @Specialization
         public DynamicObject succBang(DynamicObject string) {
             final Rope rope = rope(string);
 
             if (! rope.isEmpty()) {
                 final RopeBuilder succByteList = StringSupport.succCommon(rope);
 
-                StringOperations.setRope(string, RopeOperations.ropeFromByteList(succByteList, rope.getCodeRange()));
+                final Rope newRope = makeLeafRopeNode.executeMake(succByteList.getBytes(), rope.getEncoding(), CodeRange.CR_UNKNOWN, NotProvided.INSTANCE);
```

Comment:
Can `succ!` actually produce a different code range?

Reply:
It's strange, but possible. I had to fix this because the `succ` specs were failing.

```
x = "\xff".force_encoding('binary')
y = x.dup

x.ascii_only? # => false

y.succ! # => "\x01\x00"
y.ascii_only? # => true
```

Note that MRI currently has a bug where the code range isn't updated on `succ!`, so I had to `dup` to show the issue.

Reply:
Nice, that bug should be reported on the MRI tracker then if not already done :)